src/Microsoft.Data.SqlClient/tests/UnitTests/PoolTests.cs

```diff
@@ -459,6 +459,19 @@ public async Task ConnectionLifetime()
             Assert.That(conn.ProcessID, Is.Not.EqualTo(processId));
         }
         */
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(5)]
+        public async Task Warmup_OpenedConnectionsEqualsMinPoolSize(int minPoolSize)
+        {
+            await using var dataSource = testBase.CreateDataSource(csb => csb.MinPoolSize = minPoolSize);
+            Thread.Sleep(500*minPoolSize);
```

Comment:
Three things:
* Organizing tests into a Arrange / Act / Assert helps make them more readable what is setup tasks, what is the test action, and what needs to be verified.
* Since this test is `async`, you can use `await Task.Delay()` to sleep without blocking a thread
* However, to me having delays in a test is a code smell... imho, having arbitrary sleep times either cause tests to become flaky if the timeouts are too short or cause the test to take longer than necessary, slowing down the test run.
  * This is a complicated thing to try and avoid. My suggestion is to consider exposing a task that can be awaited. It would likely be an internal, test only, exposed object. The idea here is that the code needs to be built to be tested in a efficient manner. If you want to discuss some ideas, we can chat offline.

Reply:
I was able to rework things a bit to expose the task. This allows a direct await on the warmup task in the tests.